src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -1772,26 +1772,27 @@ internal void OnPostedOperationError(object sender, Exception e)
                         foreach (var n in names)
                         {
                             // If the types of the arguments are requested, get that from a call to -var-create
+                            string typeString = null;
                             if (types)
                             {
                                 Debug.Assert(!values, "GetParameterInfoOnly should not reach here if values is true");
-                                Results results = await MICommandFactory.VarCreate(n, thread.Id, (uint)level, 0);
-
-                                string type = results.FindString("type");
-                                args.Add(new SimpleVariableInformation(n, /*isParam*/ true, null, String.IsNullOrWhiteSpace(type) ? null : type));
-
-                                string varName = results.TryFindString("name");
-                                if (!String.IsNullOrWhiteSpace(varName))
+                                Results results = await MICommandFactory.VarCreate(n, thread.Id, (uint)level, 0, ResultClass.None);
+                                // Only get the type if the result wasn't an error
+                                if (results.ResultClass != ResultClass.error)
```

Comment:
Rather than `!= error` I think you want to check for the result class that you actually expect.